src/common/docker-compose/index.ts

```diff
@@ -75,28 +76,51 @@ export const generate = (dependency_manager: DependencyManager, build_prod = fal
           compose.services[node.normalized_ref].command = debug_options.command;
         }
 
-        // Mount the src directory
+        let volumes: string[] = [];
         const src_path = path.join(node.service_path, 'src');
         if (fs.pathExistsSync(src_path)) {
-          compose.services[node.normalized_ref].volumes = [`${src_path}:/usr/src/app/src`];
+          volumes = [`${src_path}:/usr/src/app/src`]; // Mount the src directory
         }
 
         const env_service = dependency_manager.environment.getServices()[node.ref];
+        const service_volumes = node.service_config.getVolumes();
+        const env_volumes = env_service?.debug?.volumes ? env_service.debug.volumes : {};
```

Comment:
We should always be extracting things from environment and service configs from getters/setters so we can swap out how the login works more easily.

Reply:
Currently the service config is just represented by a map, so that's why it's done this way here. In the PR for multiple interfaces though, that's updated, so I can wait until that's merged to add what you're talking about here

Reply:
All you'd have to do is add a getter to the [base](https://github.com/architect-team/architect-cli/blob/30-volumes/src/dependency-manager/src/environment-config/base.ts#L42) to get volumes. Why not just do that here?

Reply:
I didn't want to do that because rather than adding it to the base, we'll soon be able to put that on the `EnvironmentService` where it probably belongs. I've updated the PR though